subsys/input/input_longpress.c

```diff
@@ -108,35 +107,38 @@ static int longpress_init(const struct device *dev)
 	return 0;
 }
 
+/* short_codes are now optional, if not set the assert shouldn't fail */
+#define SHORT_CODES_LEN(inst) DT_PROP_LEN_OR(inst, short_codes, DT_INST_PROP_LEN(inst, input_codes))
+
 #define INPUT_LONGPRESS_DEFINE(inst)                                                               \
-	BUILD_ASSERT(DT_INST_PROP_LEN(inst, input_codes) ==                                        \
-		     DT_INST_PROP_LEN(inst, short_codes));                                         \
```

Comment:
`(DT_INST_PROP_LEN(inst, input_codes) == DT_INST_PROP_LEN(inst, short_codes)) || !DT_INST_NODE_HAS_PROP(inst, short_codes)`? No need to drop it

Reply:
Somehow I dropped this by error - readded it.

Reply:
sure, but you should still use the condition I mentioned and drop it